File: 0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int min=0,profit=0;
        for(int i=0;i<n;i++){
            if(prices[i]<prices[min])
            min=i;
            if(profit<prices[i]-prices[min])
            profit=prices[i]-prices[min];
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/solution.go
func maxProfit(prices []int) int {
    profit := 0
    currMax :=0
    for i:=len(prices)-1; i>=0; i-- {
        if profit < currMax - prices[i] {
            profit = currMax - prices[i]
        }
        if prices[i] > currMax {
            currMax = prices[i]
        }
    }
    return profit
}
